src/Symfony/Component/Security/Core/Security.php

```diff
@@ -73,6 +73,16 @@ public function isGranted(mixed $attributes, mixed $subject = null): bool
             ->isGranted($attributes, $subject);
     }
 
+    /**
+     * Checks if the attributes are granted against the user and optionally supplied subject.
+     * This should be used over isGranted() when checking permissions against a user that is not currently logged in.
+     */
+    public function userIsGranted(UserInterface $user, mixed $attributes, mixed $subject = null): bool
```

Comment:
Here the parameter is called `$attributes`, while in `UserAuthorizationCheckerInterface` it is called `$attribute` (singular) and the `UserAuthorizationChecker` wraps it in an array. Is that correct?

Reply:
This was to match the `isGranted()` function that already exists. Looking at the `AuthorizationCheckerInterface`, the parameter name was singular. I've updated both functions to use singular.

@derrabus looking at the [BC promise](https://symfony.com/doc/current/contributing/code/bc.html) updating the `isGranted()` isn't considered a BC break, but let me know if you'd like me to revert the param name change for `isGranted()`. For the new function, I agree that it makes sense to use the singular.

Reply:
I see. The PhpDoc probably also needs to be updated, as it still says "attributes".

Reply:
Updated.

Reply:
> @derrabus looking at the [BC promise](https://symfony.com/doc/current/contributing/code/bc.html) updating the param name isn't considered a BC break, but let me know if you'd like me to revert the param name change for `isGranted()`. For the new function, I agree that it makes sense to use the singular.

I'd revert that. We can debate renaming that parameter in another PR, but for this one, this is just an unnecessary and unrelated change. Please let's keep this PR focused.

Reply:
Sounds good @derrabus, on that note, would it be preferred to use singlar or plural in the new function (`userIsGranted()`)? The decision would really be around consistency vs accuracy.

Reply:
No, let's keep the singular for the new method.